src/thor/matrix_action.cc

```diff
@@ -42,10 +43,17 @@ std::string thor_worker_t::matrix(Api& request) {
                                       mode, max_matrix_distance.find(costing)->second);
   };
   auto timedistancematrix = [&]() {
-    return time_distance_matrix_.SourceToTarget(options.sources(), options.targets(), *reader,
-                                                mode_costing, mode,
-                                                max_matrix_distance.find(costing)->second,
-                                                options.matrix_locations());
+    time_distances =
+        forward_search
+            ? time_dist_matrix_forward_.SourceToTarget(options.sources(), options.targets(), *reader,
+                                                       mode_costing, mode,
+                                                       max_matrix_distance.find(costing)->second,
+                                                       options.matrix_locations())
+            : time_dist_matrix_reverse_.SourceToTarget(options.sources(), options.targets(), *reader,
+                                                       mode_costing, mode,
+                                                       max_matrix_distance.find(costing)->second,
+                                                       options.matrix_locations());
+    return time_distances;
```

Comment:
either forward or reverse, depending if there's more sources or targets. I'd like to have said `td_algo = forward_search ? time_dist_matrix_forward_ : time_dist_matrix_reverse_`, but it gave an error about `deleted function`. if anyone has the c++ sugar handy to make that happen (if possible), I'd appreciate it

Reply:
never mind, this is a stupid pattern, see https://github.com/valhalla/valhalla/pull/3773#discussion_r985280041